packages/flutter/test/widgets/basic_test.dart

```diff
@@ -0,0 +1,128 @@
+        const Directionality(
+          textDirection: TextDirection.rtl,
+          child: const PhysicalShape(
+            clipper: const ShapeBorderClipper(shapeBorder: const CircleBorder()),
+            color: const Color(0xFF0000FF),
+            textDirection: TextDirection.ltr,
+          ),
+        ),
+      );
+      final RenderPhysicalShape renderObject = tester.renderObject(find.byType(PhysicalShape));
+      expect(renderObject.textDirection, TextDirection.ltr);
+    });
+
+    testWidgets('inherits directionality', (WidgetTester tester) async {
+      await tester.pumpWidget(
+        const Directionality(
+          textDirection: TextDirection.rtl,
+          child: const PhysicalShape(
+            clipper: const ShapeBorderClipper(shapeBorder: const CircleBorder()),
+            color: const Color(0xFF0000FF),
+          ),
+        ),
+      );
+      final RenderPhysicalShape renderObject = tester.renderObject(find.byType(PhysicalShape));
+      expect(renderObject.textDirection, TextDirection.rtl);
+    });
+
+    testWidgets('hit test', (WidgetTester tester) async {
+      await tester.pumpWidget(
+        new PhysicalShape(
+          clipper: const ShapeBorderClipper(shapeBorder: const CircleBorder()),
+          elevation: 2.0,
+          color: const Color(0xFF0000FF),
+          shadowColor: const Color(0xFF00FF00),
+          child: new Container(color: const Color(0xFF0000FF)),
+        )
+      );
+
+      final RenderPhysicalShape renderPhysicalShape =
+        tester.renderObject(find.byType(PhysicalShape));
+
+      // The viewport is 800x600, the CircleBorder is centered and fits
+      // the shortest edge, so we get a circle of radius 300, centered at
+      // (400, 300).
+      //
+      // We test by sampling a few points around the left-most point of the
+      // circle (100, 300).
+
+      expect(tester.hitTestOnBinding(const Offset(99.0, 300.0)), doesntHit(renderPhysicalShape));
+      expect(tester.hitTestOnBinding(const Offset(100.0, 300.0)), hits(renderPhysicalShape));
+      expect(tester.hitTestOnBinding(const Offset(100.0, 299.0)), doesntHit(renderPhysicalShape));
+      expect(tester.hitTestOnBinding(const Offset(100.0, 301.0)), doesntHit(renderPhysicalShape));
+    });
+
+  });
+
+}
+
+HitsRenderBox hits(RenderBox renderBox) => new HitsRenderBox(renderBox);
+
+class HitsRenderBox extends Matcher {
+  const HitsRenderBox(this.renderBox);
+
+  final RenderBox renderBox;
+
+  @override
+  Description describe(Description description) =>
+    description.add('hit test result contains ').addDescriptionOf(renderBox);
+
+  @override
+  bool matches(dynamic item, Map<dynamic, dynamic> matchState) {
+    final HitTestResult hitTestResult = item;
+    return hitTestResult.path.where(
+      (HitTestEntry entry) => entry.target == renderBox
+    ).isNotEmpty;
+  }
+}
+
+DoesntHitRenderBox doesntHit(RenderBox renderBox) => new DoesntHitRenderBox(renderBox);
+
+class DoesntHitRenderBox extends Matcher {
```

Comment:
trivial nit: "DoesNot" rather than "Doesnt" to avoid the awkward missing apostrophe

Reply:
done